hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/ethereum/EthereumSuite.java

```diff
@@ -905,6 +907,46 @@ HapiSpec transferTokensViaEip2930TxSuccessfully() {
                                                         .withErcFungibleTransferStatus(true)))))));
     }
 
+    @HapiTest
+    HapiSpec overflowFromGasPriceTimesGasLimitFailsAppropriately() {
+        return defaultHapiSpec("overflowFromGasPriceTimesGasLimitFailsAppropriately")
+                .given(
+                        withOpContext((spec, ctxLog) -> spec.registry().saveContractId("invalid", asContract("1.1.1"))),
```

Comment:
Do we need to be calling an invalid contract for this scenario? 
It's no big deal, but it's a bit confusing IMO because it mixes a couple different test scenarios. It would be more straightforward if we just attempt to deploy a small/empty contract or something similar, any operation that would be otherwise successful, except for the gas price * gas cost overflow.

Reply:
It seemed easier to do it this way since - with knowledge of what's going on inside - this check happens _before_ any contract is called (since it is validating fees are available).

But you're right, it is confusing, and I'll fix it.